common/src/main/java/io/druid/math/expr/ExprEval.java

```diff
@@ -99,10 +98,10 @@ public Object value()
     return value;
   }
 
-  public boolean isNull()
```

Comment:
this method is replaced with isNumericNull as that is what we need to check for Expressions. 
isNumericNull will check whether the primitive long/float/double equivalent is null or not. 
It is possible for an expression to have a non-null String value but it can return null when parsed as a primitive long/float/double.